blockchain/chainview.go

```diff
@@ -12,6 +12,25 @@ import (
 // in a week on average.
 const approxNodesPerWeek = 6 * 24 * 7
 
+// log2FloorMasks defines the masks to use when quickly calculating
+// floor(log2(x)) in a constant log2(32) = 5 steps, where x is a uint32, using
+// shifts.  They are derived from (2^(2^x) - 1) * (2^(2^x)), for x in 4..0.
+var log2FloorMasks = []uint32{0xffff0000, 0xff00, 0xf0, 0xc, 0x2}
+
+// fastLog2Floor calculates and returns floor(log2(x)) in a constant 5 steps.
+func fastLog2Floor(n uint32) uint8 {
```

Comment:
There isn't one.  This really is only used in this one place though (it looks like an addition because I moved it out of `blocklocator.go`), so I don't think creating a separate package for just this one use is worth it here.